### ISSUE TITLE
Same test gives different results for 1.5 rke permissive vs 1.6 rke permissive scan on the same cluster

### ISSUE THREAD

**Issue by sowmyav27** [rancher/cis-operator#62]
On master-head - commit id: `e2e530c19a`

- run 1.5 permissive and 1.6 permissive
- one test for example 1.1.12
**1.5 permissive:**
```
{"id":"1.1.12","description":"Ensure
    that the etcd data directory ownership is set to etcd:etcd (Scored)","remediation":"On
    the etcd server node, get the etcd data directory, passed as an argument --data-dir,\nfrom
    the below command:\nps -ef | grep etcd\nRun the below command (based on the etcd
    data directory found above).\nFor example, chown etcd:etcd /var/lib/etcd\n\nA
    system service account is required for etcd data directory ownership.\nRefer to
    Rancher''s hardening guide for more details on how to configure this ownership.\n","state":"notApplicable","node_type":["etcd"],"audit":"stat
    -c %U:%G /node/var/lib/etcd","audit_config":"","test_info":null,"commands":null,"config_commands":null,"actual_value":"","expected_result":"","test_type":"skip","scored":true}]}
```
**1.6 permissive:**
```
{"id":"1.1.12","description":"Ensure
    that the etcd data directory ownership is set to etcd:etcd (Automated)","remediation":"On
    the etcd server node, get the etcd data directory, passed as an argument --data-dir,\nfrom
    the below command:\nps -ef | grep etcd\nRun the below command (based on the etcd
    data directory found above).\nFor example, chown etcd:etcd /var/lib/etcd\n","state":"pass","node_type":["master"],"audit":"ps
    -ef | grep $etcdbin | grep -- --data-dir | sed ''s%.*data-dir[= ]\\([^ ]*\\).*%\\1%''
    | xargs stat -c %U:%G","audit_config":"","test_info":null,"commands":null,"config_commands":null,"actual_value":"","expected_result":"","test_type":"","scored":true}
```

**Question:**
Why is the same test run on the same cluster have different outputs ? Is that expected ?

**Comment by sowmyav27** [comment]
Related issue - https://github.com/rancher/rancher/issues/29649#issuecomment-733985659

**Comment by mrajashree** [comment]
Tests 1.1.11, 1.1.12 1.1.19, 1.1.20 and 1.1.21 are a part of master.yaml as per CIS kube-bench upstream profiles, but security-scan tool needs to run them on etcd nodes since they check permissions for etcd. These were a part of etcd.yaml in RKE 1.5 profiles so they were being run on etcd nodes for those profiles. But for RKE 1.6 they were still a part of master.yaml, hence being run on node_type master. All 5 tests are now moved to etcd.yaml
From this test 1.1.12 can pass on a hardened cluster but not on a non-hardened, so it is skipped in permissive profiles (was already skipped in 1.5, and now will be skipped in 1.6 too)

**Comment by sowmyav27** [comment]
@mrajashree Can you confirm the behavior for the following tests

- 1.1.11 and 1.1.12 are seen as node_type - `etcd` for both rke 1.5 and 1.6 permissive scan runs
- 1.1.19, 1.1.20 and 1.1.21, however, are seen as `etcd` for rke 1.6 permissive scan run and `etcd`, `master` for rke 1.5 permissive scan run.

[Archive.zip](https://github.com/rancher/cis-operator/files/5683710/Archive.zip)

The tests 
<img width="1414" alt="Screen Shot 2020-12-12 at 10 54 41 AM" src="https://user-images.githubusercontent.com/26032343/101994182-343a2c80-3c75-11eb-8d16-a7aafa6058b3.png">
now have the same set of pass/n/a though. - latest 2.5-head using latest CIS v2 charts

**Comment by mrajashree** [comment]
Available to test with v0.2.2-rc4 of security-scan
Tests 1.1.19, 1.1.20 and 1.1.21 will be run on both, etcd and master nodes since those tests are for certs and both node types have certs

**Comment by sowmyav27** [comment]
**On 2.5-head commit id: `182e3de`**

- CIS v 1.0.3 and v0.2.2-rc4 of security-scan
- 1.1.19, 1.1.20 and 1.1.21 show both etcd and master for all CIS 1.5, 1.6 permissive and hardened cluster scans
